Go unit and integration tests and Dockerfile (#4625)
awsdocs/aws-doc-sdk-examples: gov2/README.md

go test -tags=integration ./...
```

> ## Docker image

Reviewer: ```suggestion
## Docker image (Beta)
```

Reviewer: We need this permalink to remain consistent. Referenced in ECR.

This example code will soon be available in a container image
hosted on [Amazon Elastic Container Registry (Amazon ECR)](https://docs.aws.amazon.com/AmazonECR/latest/userguide/what-is-ecr.html). 
The image will be preloaded with all Go examples, with dependencies pre-resolved. 
That way, you can explore the examples in an isolated environment.

> ⚠️ As of January 2023, the [SDK for Go V2 image](https://gallery.ecr.aws/b4v4v1s0/gov2) is available on ECR Public 
> but is still undergoing active development. Refer to [this GitHub issue](https://github.com/awsdocs/aws-doc-sdk-examples/issues/4125)
> for more information.

Reviewer 2: ```suggestion
⚠️ As of January 2023, the [SDK for Go V2 image](https://gallery.ecr.aws/b4v4v1s0/gov2) is available in the Amazon ECR Public Gallery but is still undergoing active development. Refer to [this GitHub issue](https://github.com/awsdocs/aws-doc-sdk-examples/issues/4125)
for more information.
```

### Build the Docker image

1. Install and run Docker on your machine.
2. Navigate to the same directory as this README.
3. Run `docker build -t <image_name> .` and replace `image_name` with a name for the image.

### Launch the Docker container

> Run the Docker container with your image with the following command:

Reviewer 2: ```suggestion
Run the Docker container with your image using the following command:
```

**Windows**

```
docker run -it --volume %USERPROFILE%\.aws:/root/.aws <image_name>
```

**macOS or Linux**
```
docker run -it -v ~/.aws/credentials:/root/.aws/credentials <image_name>
```

The terminal initiates a bash instance at the root of the container.
The Go code examples are in the `gov2` folder and can be run by following
the instructions in the READMEs in the various folders.

### Run tests in the Docker container

> You can run all unit tests and write the output to a file by running the following command

Reviewer 2: ```suggestion
To run all unit tests and write the output to a file, run the following command
```

in the `gov2` directory of the container:  

```
./run_all_tests.sh
```

> You can run integration tests by passing an `integration` argument to the `run_all_tests.sh` 

Reviewer 2: ```suggestion
To run integration tests, pass an `integration` argument to the `run_all_tests.sh` 
```
